libbeat/publisher/queue/diskqueue/config.go

```diff
@@ -68,6 +71,16 @@ type Settings struct {
 	// use exponential backoff up to the specified limit.
 	RetryInterval    time.Duration
 	MaxRetryInterval time.Duration
+
+	// Schema Version specifies which on-disk format, serialization, and encryption to use.
+	// 0, 1, or 2 are valid options
+	SchemaVersion uint32
```

Comment:
Schema version shouldn't be a configurable option, as we don't actually want to support multiple versions, we just need to be able to read old data after a user update. So we should be able to read old versions of the schema (though we might at some point remove 0, since it only appeared in beta), but new segments should always be written in the most recent version. (This also means we can assume schema-2 fields are present on any segments we create, and just need to pick reasonable defaults for them when loading an older version.)

Reply:
> Schema version shouldn't be a configurable option, as we don't actually want to support multiple versions, we just need to be able to read old data after a user update. So we should be able to read old versions of the schema (though we might at some point remove 0, since it only appeared in beta), but new segments should always be written in the most recent version. (This also means we can assume schema-2 fields are present on any segments we create, and just need to pick reasonable defaults for them when loading an older version.)

So this gets back to the feedback from the design doc.  The feedback was to have version 3 be encrypted, and not have fields in the segment header specifying things like encrypted, ciphers, keys, etc.  So **if** version 3 is always encrypted, the option exists so you can choose the non-encrypted version 2.

That being said, the original design had more fields in the segment header to control things like encryption, I'd be very happy to go back towards that.